cmd/subctl/recover_broker_info.go

```diff
@@ -63,114 +55,57 @@ func init() {
 	rootCmd.AddCommand(recoverBrokerInfo)
 }
 
-func recoverBrokerFromConfigContext(brokerCluster *cluster.Info, brokerNamespace string, status reporter.Interface) error {
-	brokerObj, err := getBroker(brokerCluster, brokerNamespace, "Please try again with another cluster", status)
-	if err != nil {
-		return err
-	}
+func recoverBrokerInfoFromSubm(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+	brokerNamespace := submCluster.Submariner.Spec.BrokerK8sRemoteNamespace
+	brokerRestConfig := submCluster.RestConfig
 
-	clusters, err := brokerCluster.GetClusters(brokerNamespace)
+	status.Start("Checking if the Broker is installed on the Submariner cluster %q in namespace %q", submCluster.Name, brokerNamespace)
+
+	brokerObj, err := getBroker(brokerRestConfig, brokerNamespace)
 	if err != nil {
-		return status.Error(err, "error listing joined clusters")
+		status.Warning("Error getting Broker: %s", err)
 	}
 
-	ok, err := tryToRecoverFromBroker(brokerCluster, brokerObj, brokerNamespace, clusters, status)
-	if ok || err != nil {
-		return err
+	if brokerObj == nil {
+		status.Start("Trying to connect to the Broker installed on a different cluster")
+
+		brokerRestConfig, brokerNamespace, err = restconfig.ForBroker(submCluster.Submariner, submCluster.ServiceDiscovery)
+		if err != nil || brokerRestConfig == nil {
+			return status.Error(err, "Error getting the broker's rest config")
+		}
+
+		brokerObj, err = getBroker(brokerRestConfig, brokerNamespace)
+		if err != nil {
+			return status.Error(err, "Error getting Broker")
```

Comment:
The error already contains enough info.

```suggestion
			return status.Error(err, "")
```